src/support/mod.rs

```diff
@@ -137,12 +138,49 @@ pub fn get_llvm_version() -> (u32, u32, u32) {
     return (major, minor, patch);
 }
 
+/// Permanently load the dynamic library with the given `filename`.
+///
+/// It is safe to call this function multiple times for the same library.
+///
+/// Returns `true` if there was an error while loading the library.
+#[deprecated(note = "Use `try_load_library_permanently` instead")]
 pub fn load_library_permanently(filename: &str) -> bool {
     let filename = to_c_str(filename);
 
     unsafe { LLVMLoadLibraryPermanently(filename.as_ptr()) == 1 }
 }
 
+/// Possible errors that can occur when loading a library
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+pub enum LoadLibraryError {
+    /// The given path could not be converted to a [`&str`]
+    UnicodeError,
+    /// The given path could not be loaded as a library
+    LoadingError,
+}
+
+/// Permanently load the dynamic library at the given `path`.
+///
+/// It is safe to call this function multiple times for the same library.
```

Comment:
Did you get this from the docs somewhere? How are we sure it's safe?

Reply:
https://llvm.org/doxygen/classllvm_1_1sys_1_1DynamicLibrary.html#a53d32d3b3baefdec31d3d94b0586d437

Reply:
Or for C wrapper: https://llvm.org/doxygen/group__LLVMCCore.html#ga6fc1331c1a6d2cc6f0fda94f4a4636f9